src/mono/mono/mini/interp/transform.c

```diff
@@ -2445,6 +2445,41 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 				has_refs = m_class_has_references (klass);
 
 			*op = has_refs ? MINT_LDC_I4_1 : MINT_LDC_I4_0;
+		} else if (!strcmp (tm, "CreateSpan") && csignature->param_count == 1 && td->last_ins->opcode == MINT_LDPTR) {
+			MonoGenericContext* ctx = mono_method_get_context (target_method);
+			g_assert (ctx);
+			g_assert (ctx->method_inst);
+			g_assert (ctx->method_inst->type_argc == 1);
+			MonoType *span_arg_type = mini_get_underlying_type (ctx->method_inst->type_argv [0]);
+
+			MonoClassField *field = (MonoClassField*)td->data_items [td->last_ins->data [0]];
+			int alignment = 0;
```

Comment:
Is it guaranteed that td->last_ins comes from LDTOKEN ?

Reply:
Double checked that this condition handles all the cases of `CreateSpan` invocation from `System.Runtime.Tests` suite. Doesn't seem possible as of right now, but nevertheless made the condition more strict to prevent the case where `MINT_LDPTR` would be the last instruction but not really the argument to `CreateSpan`.